src/misc/stringToNumber.ts

```diff
@@ -0,0 +1,9 @@
+/** Returns a numeric representation of an input string.
+ * The string represents one or more digits, with exactly one
+ * optional decimal place, at any location in the string.
+ * This may return NaN.
+ * @param str String that represents a number, optionally with one decimal place.
+ * */
+export const stringToNumber = (str: string): number => (
+  /^((\d+(\.\d*)?)|\.\d+)$/.test(str) ? parseFloat(str) : NaN
+);
```

Comment:
@brendandagys I believe having one space at the end of the file is part of the linting rules. Did you apply eslint with the rules used by the project's `.eslintrc` to your submission?

Reply:
I read a while ago that having an empty line at the end of a document is a good practice as you suggest, and I strive to adhere to that. I double-checked the code that I checked in, and there is in fact a blank line at the end of the file.

I believe that GitHub is truncating it. Looking at other files on GitHub in your library show the omission of it as well, so I believe this is a UI quirk.